ferrous-game/Assets/Scripts/Mechanics/SwitchManager.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.Events;
+using System;
+
+namespace Ferrous.Mechanics
+{
+    public class SwitchManager : BaseSwitch
+    {
+        [SerializeField] private UnityEvent onAllSwitchesActivated;
+        [SerializeField] private List<BaseSwitch> activatedSwitches = new List<BaseSwitch>();
+        private bool allActivated = false;
+
+
+        void Update()
+        {
+            DetectActive(activatedSwitches);
```

Comment:
i think the implementation `DetectActive(activatedSwitches)` should go in an `ActivateObject()` implementation

you don't need to pass in `activatedSwitches`, just get the component in `void Start()`